sdk/src/wallet/types/balance.rs

```diff
@@ -23,43 +23,43 @@ pub struct Balance {
     /// Current required storage deposit amount
     pub(crate) required_storage_deposit: RequiredStorageDeposit,
     /// Native tokens
-    pub(crate) native_tokens: Vec<NativeTokensBalance>,
+    pub(crate) native_tokens: BTreeMap<TokenId, NativeTokensBalance>,
     /// Accounts
-    pub(crate) accounts: Vec<AccountId>,
+    pub(crate) accounts: BTreeSet<AccountId>,
     /// Foundries
-    pub(crate) foundries: Vec<FoundryId>,
+    pub(crate) foundries: BTreeSet<FoundryId>,
     /// Nfts
-    pub(crate) nfts: Vec<NftId>,
+    pub(crate) nfts: BTreeSet<NftId>,
     /// Delegations
-    pub(crate) delegations: Vec<DelegationId>,
+    pub(crate) delegations: BTreeSet<DelegationId>,
     /// Outputs with multiple unlock conditions and if they can currently be spent or not. If there is a
     /// [`TimelockUnlockCondition`](crate::types::block::output::unlock_condition::TimelockUnlockCondition) or
     /// [`ExpirationUnlockCondition`](crate::types::block::output::unlock_condition::ExpirationUnlockCondition) this
     /// can change at any time
-    pub(crate) potentially_locked_outputs: HashMap<OutputId, bool>,
+    pub(crate) potentially_locked_outputs: BTreeMap<OutputId, bool>,
+    pub(crate) potential_mana: u64,
+    pub(crate) stored_mana: u64,
+    pub(crate) block_issuance_credits: BTreeMap<AccountId, i128>,
```

Comment:
Maybe something like this
```suggestion
    /// Decayed potential Mana of owned outputs.
    pub(crate) potential_mana: u64,
    /// Decayed stored Mana of owned outputs.
    pub(crate) stored_mana: u64,
    /// Block issuance credits of wallet accounts.
    pub(crate) block_issuance_credits: BTreeMap<AccountId, i128>,
```
also compared to the base coin these values would add up to the `total Mana`, are we maybe also interested in `available Mana`?

Reply:
Yeah I'd maybe even put that in a new struct like BaseCoin